CDCgov/prime-reportstream
# CA Flu Condition Filters Proposal

This PR is a proposal for the CA Flu condition filter. It also adds a development organization to the organizations.yml file. The express purpose of this organization is for local development.  Within that organization, I've created a full elr development receiver with my proposal.  The list of cities within LA County and Orange County have been added to the fhirpath_filter_shorthand table.

CA wants to receive:
- Positive Flu A results
- Positive Flu B results
- Negative Flu A results where the facility or patient is in LA County or Orange County
- Negative Flu B results where the facility or patient is in LA County or Orange County


**If you are suggesting a fix for a currently exploitable issue, please disclose the issue to the prime-reportstream team directly outside of GitHub instead of filing a PR, so we may immediately patch the affected systems before a disclosure. See [SECURITY.md/Reporting a Vulnerability](https://github.com/CDCgov/prime-reportstream/blob/master/SECURITY.md#reporting-a-vulnerability) for more information.**

Test Steps:
Since there are changes to both a lookup table and organizations.yml, you must `./gradlew reloadTables` and `./gradlew reloadSettings` before beginning this process.

1. Send a POST request to send a bulk FHIR report. Sample request:
[CA Flu Condition FIlter Test.postman_collection.txt](https://github.com/CDCgov/prime-reportstream/files/11045189/CA.Flu.Condition.FIlter.Test.postman_collection.txt)

2. Verify that the request was accepted

3. (Optional) Call the submission history API to verify the reports were delivered: [http://localhost:7071/api/waters/report/{{reportId}}/history](http://localhost:7071/api/waters/report/%7B%7BreportId%7D%7D/history)

4. Use Azure Storage Explorer and look in the batch folder to see that the message was routed properly or Check the local test sftp folder.
5. Ensure that only the positive Flu B observation is sent.
6. Change the patient or performer address in the FHIR bundle to be in one of the cities that want to receive negatives (ex: `Anaheim`)
7. Re run steps 1-4. This time both the negative flu A and the positive flu B should be in the final HL7 message

## Changes
- Added Development Organization in `organizations.yml`
- Added Flu Condition Filter
- Added fhirpath shorthands for shorter more comprehensible expressions.

## Checklist

### Testing

### Process
- [ ] This makes changes to a lookup table. Staging and prod lookup tables need to be updated before this deploys

## Linked Issues
- #8587 

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?
